File: data-structures/javascript/src/sort/insertion.js
function insertionSort(array) {
  for (let index = 1; index < array.length; index++) {
    value = array[index]

    previousValue = index - 1

    while (previousValue >= 0 && array[previousValue] > value) {
      array[previousValue + 1] = array[previousValue]
      previousValue = previousValue - 1
    }

    array[previousValue + 1] = value
  }
}

function insertionInvertSort(array) {
  for (let index = 1; index < array.length; index++) {
    value = array[index]

    previousValue = index - 1

    while (previousValue >= 0 && array[previousValue] < value) {
      array[previousValue + 1] = array[previousValue]
      previousValue = previousValue - 1
    }

    array[previousValue + 1] = value
  }
}

let arr = [1, 3, 4, 10, 25, 11, 5, 2]
let arrInvert = [1, 3, 4, 10, 25, 11, 5, 2]

insertionSort(arr)
insertionInvertSort(arrInvert)

console.log(arr)
console.log(arrInvert)

File: data-structures/python/src/sort/insertion.py
def insertionSort(array):
    for index in range(1, len(array)):
        value = array[index]
        previousIndex = index - 1
        while previousIndex >= 0 and array[previousIndex] > value:
            array[previousIndex + 1] = array[previousIndex]
            previousIndex = previousIndex - 1
        array[previousIndex + 1] = value
    return


def insertionInvertSort(array):
    for index in range(1, len(array)):
        value = array[index]
        previousIndex = index - 1
        while previousIndex >= 0 and array[previousIndex] < value:
            array[previousIndex + 1] = array[previousIndex]
            previousIndex = previousIndex - 1
        array[previousIndex + 1] = value
    return


arr = [5, 3, 11, 2]
insertionSort(arr)

print(arr)
